akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileUploadDirectives.scala

```diff
@@ -58,6 +59,37 @@ trait FileUploadDirectives {
     }
 
   /**
+   * Streams the bytes of the file submitted using multipart with the given field name into temporary files on disk.
+   * If there is an error writing to disk the request will be failed with the thrown exception, if there is no such
+   * field the request will be rejected. Stored files are cleaned up on exit but not on failure.
+   *
+   * @group fileupload
+   */
+  def uploadedFiles(fieldName: String): Directive1[immutable.Seq[(FileInfo, File)]] =
+    entity(as[Multipart.FormData]).flatMap { formData ⇒
+      extractRequestContext.flatMap { ctx ⇒
+        implicit val mat = ctx.materializer
+        implicit val ec = ctx.executionContext
+
+        val uploaded: Source[(FileInfo, File), Any] = formData.parts
+          .filter(part ⇒ part.filename.isDefined && part.name == fieldName)
+          .mapAsync(1) { part ⇒
+            val fileInfo = FileInfo(part.name, part.filename.get, part.entity.contentType)
+            val dest = File.createTempFile("akka-http-upload", ".tmp")
```

Comment:
Maybe this part should be completely abstracted away, so that the user can choose the destination freely based on the file info.